src/collection.cpp

```diff
@@ -5025,3 +5030,40 @@ Option<bool> Collection::truncate_after_top_k(const string &field_name, size_t k
 
     return Option<bool>(true);
 }
+
+void Collection::remove_embedding_field(const std::string& field_name) {
+    field del_field;
+
+    if(embedding_fields.find(field_name) != embedding_fields.end()) {
```

Comment:
You can do early return with `==` check so that `del_field` can be directly assigned after that.